index.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+// See https://github.com/jshint/jshint/issues/1747 for context
+/* global -Promise */
+var Promise = require('es6-promise').Promise;
+var stream = require('stream');
+
+module.exports = function(options) {
+
+  options = options || {};
+
+  var client = options.client;
+  var manifest = options.manifest;
+
+  var webAppsActor;
+  return getWebAppsActor(client).then(function(result) {
+
+    webAppsActor = result;
+
+    return Promise.all([
+      webAppsActor.getInstalledApps(),
+      webAppsActor.listRunningApps()
+    ]);
+
+  }).then(function(results) {
+
+    var installedApps = results[0];
+    var runningAppIds = results[1];
+
+    // Find installed & running apps that match the manifest
+    var selectedRunningAppIds = installedApps.filter(function(app) {
+      return app.name === manifest.name &&
+             runningAppIds.indexOf(app.manifestURL) !== -1;
+    }).map(function(app) {
+      return app.manifestURL;
+    });
+    if (selectedRunningAppIds.length === 0) {
+      throw new Error('app not installed and running');
+    }
+
+    // Finally, get a console for the first installed & running app.
+    return webAppsActor.getApp(selectedRunningAppIds[0]);
+
+  }).then(function(app) {
+
+    // Wrap the useful methods of Console in promise handlers
+    var wrapped = promisifyMethods(app.Console, [
+      'startListening',
+      'stopListening',
+      'evaluateJS',
+      'getCachedLogs',
+      'clearCachedLogs'
+    ]);
+
+    // Enhance by injecting log stream helper
+    wrapped.getLogStream = getLogStream.bind(wrapped);
+
+    return wrapped;
+
+  });
+
+};
+
+// Method to wrap a stream around console log events.
+function getLogStream(options) {
+  options = options || { logs: true };
+  /*jshint validthis:true */
```

Comment:
[validthis](http://jshint.com/docs/options/#validthis) is a scope-specific directive, so we can't use it globally. Basically, this looks weird to jshint because I'm using `this` in something that doesn't look like an object method, but the directive says this particular case is okay